app.js

```diff
@@ -1,63 +1,219 @@
+    options[option] = req.query[option];
+  });
+
+  if (req.instance) {
+    options.branchName = keypather.get(req.instance, 'attrs.contextVersion.branch');
+    // Temp missing pending resolution of SAN-3018
+    // https://runnable.atlassian.net/browse/SAN-3018
+    options.instanceName = keypather.get(req.instance, 'attrs.contextVersion.branch');
+    options.ownerName = keypather.get(req.instance, 'attrs.owner.username');
+    var ports = keypather.get(req.instance, 'attrs.container.ports');
+    options.ports = Object.keys(ports).map(function (portKey) {
+      // Ex: '3000/tcp' --> '3000'
+      return portKey.replace(/\/tcp$/, '');
     });
-    if (req.query.ports) {
-      var value = req.query.ports;
-      if (!Array.isArray(value)) {
-        value = [value];
-      }
-      options.ports = value;
-    }
-    options.headerText = options.status;
-    if (options.status) {
-      if (options.status === 'buildFailed') {
-        options.headerText = 'build failed.';
-        options.status = 'failed to build';
-      } else {
-        options.status = 'is ' + options.status;
-      }
+  }
+
+  if (req.query.type === 'signin') {
+    log.trace('processNaviError type signin');
+    return res.render('pages/signin', options);
+  } else if (req.query.type === 'not_running') {
+    log.trace('processNaviError type not_running');
+
+    // container state error pages.
+    // - Not running (building, starting, crashed)
+    // - Running, but unresponsive
+    var status = req.instance.status();
+    log.trace({
+      status: status,
+      options: options
+    }, 'processNaviError instance status');
+
+    options.status = status;
+    switch(status) {
+      case 'stopped':
+      case 'crashed':
+      case 'stopping':
+        options.headerText = 'is ' + status;
+        res.render('pages/dead', options);
+        break;
+      case 'running':
+        // The instance could have started after Navi fetched it and proxied to detention.
+        // Might not be the best idea to trigger a refresh, could easily result in user-unfriendly
+        // infinite redirect loops. Better to display an error page prompting user to refresh?
+        options.headerText = 'is running';
+        res.render('pages/dead', options);
+        break;
+      case 'buildFailed':
+        options.headerText = 'build failed';
+        res.render('pages/dead', options);
+        break;
+      case 'building':
+        options.headerText = 'is building';
+        res.render('pages/dead', options);
+        break;
+      case 'neverStarted':
+      case 'starting':
+        options.headerText = 'is starting';
+        res.render('pages/dead', options);
+        break;
+      case 'unknown':
+        options.headerText = 'unknown';
+        res.render('pages/dead', options);
+        break;
     }
-    res.render('pages/' + page, options);
-  } else {
-    res.render('pages/invalid', options);
+  } else if (req.query.type === 'ports'){
```

Comment:
not really a fan of these long running `else if's` have to keep scrolling up to see which chain it is a part of. Would it be hard converting this into returns?